go/ql/lib/semmle/go/security/CleartextLogging.qll

```diff
@@ -56,4 +58,43 @@ module CleartextLogging {
       not any(Protobuf::GetMethod gm).taintStep(src, trg)
     }
   }
+
+  private module Config implements DataFlow::ConfigSig {
+    predicate isSource(DataFlow::Node source) { source instanceof Source }
+
+    predicate isSink(DataFlow::Node sink) { sink instanceof Sink }
+
+    predicate isBarrier(DataFlow::Node node) {
+      node instanceof Barrier
+      or
+      exists(DataFlow::CallNode call | node = call.getResult() |
+        call.getTarget() = Builtin::error().getType().getMethod("Error")
+        or
+        call.getTarget().(Method).hasQualifiedName("fmt", "Stringer", "String")
+      )
+    }
+
+    predicate isAdditionalFlowStep(DataFlow::Node src, DataFlow::Node trg) {
+      // A taint propagating data-flow edge through structs: a tainted write taints the entire struct.
+      exists(Write write |
+        write.writesField(trg.(DataFlow::PostUpdateNode).getPreUpdateNode(), _, src)
+      )
+      or
+      // taint steps that do not include flow through fields. Field reads would produce FPs due to
+      // the additional taint step above that taints whole structs from individual field writes.
+      TaintTracking::localTaintStep(src, trg) and
+      not TaintTracking::fieldReadStep(src, trg) and
+      // Also exclude protobuf field fetches, since they amount to single field reads.
+      not any(Protobuf::GetMethod gm).taintStep(src, trg)
+    }
+  }
+
+  /**
+   * Tracks data flow for reasoning about clear-text logging of sensitive
+   * information, from `Source`s, which are sources of sensitive data, to
+   * `Sink`s, which is an abstract class representing all the places sensitive
+   * data may be stored in cleartext. Additional sources or sinks can be added
+   * by extending the relevant class.
```

Comment:
This is copied from elsewhere in the file. I'm sure there are many improvements we could make to our documentation, but I suggest this could be done in a separate PR afterwards.